webpages/api/admin/module_model.php

```diff
@@ -26,32 +26,56 @@ public function getDescriptorFileName() {
         return "/module/" . str_replace('.', '/', $this->packageName) . "_module.php";
     }
 
-    // Some modules have a "descriptor" to help describe the full feature set
-    // of a module. The Descriptor is implemented in a class with a special
-    // naming pattern.
-    public function getDescriptorClassName() {
-        $index = strpos($this->packageName, ".");
-        if ($index > 0) {
-            $namespaceName = substr($this->packageName, 0, $index);
-            if ($namespaceName == "planz") {
-                $namespaceName = 'PlanZ\Module\\';
-            } else {
-                $namespaceName = ucfirst(mb_strtolower($namespaceName)) . '\\';
-            }
-
-            $temp = substr($this->packageName, $index + 1);
-            $className = "";
-            while (strpos($temp, '_') !== false) {
-                $index = strpos($temp, '_');
-                $className .= ucfirst(mb_strtolower(substr($temp, 0, $index)));
-                $temp = substr($temp, $index + 1);
-            }
-            $className .= ucfirst(mb_strtolower($temp));
-
-            return $namespaceName . $className . "Module";
-        } else {
+    /**
+     * Some modules have a "descriptor" to help describe the full feature set of
+     * a module. The Descriptor is implemented in a class with a special naming
+     * pattern.
+     *
+     * @return string
+     */
+    public function getDescriptorClassName(): string
+    {
+        $names = explode('.', $this->packageName);
+        if (count($names) < 2) {
             return null;
         }
+        $namespaceName = $this->makeNameSpace($names[0]);
+        $className = $this->makePascalCase($names[1]);
+        return $namespaceName . $className . "Module";
+    }
+
+    /**
+     * Return a PHP namespace from the package name.
+     *
+     * @param string $name The snake_case name to make namespace from.
+     *
+     * @return string
+     */
+    protected function makeNameSpace(string $name): string
+    {
+        if ($name == 'planz') {
+            return 'PlanZ\Module\\';
+        }
+        // Split package prefix into words, and capitalize each word.
+        return 'PlanZ\\' . $this->makePascalCase($name) . '\\';
```

Comment:
I think this represents a change in behaviour. I have some WisCon-specific modules that use "Wiscon" as the namespace; I wouldn't have expected "PlanZ" to be part of the namespace.

Reply:
Thanks. I hadn't considered modules from outside the PlanZ project. I will look at this again.
Can you give an example of a package name and the resulting namespace for one of the WisCon modules?

Reply:
Sure. I have a module that creates the JSON extract for the WisSched iOS and Android apps (kinda-sorta comparable to what you do with ConClàr). 

The database package name is "wiscon.wissched", and the module definition is in module/wiscon/wissched_module.php with a namespace of "Wiscon".

Reply:
I've revised, so it should now create the same namespace name.

The case where the namespace name would not be the same is where the first part of the package name contains underscores.

For example, `table_tents` would previously have produced the namespace `Table_tents`, but now produces `TableTents`.